sky/packages/sky/lib/rendering/object.dart

```diff
@@ -396,6 +479,16 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     assert(node._relayoutSubtreeRoot == node);
     return true;
   }
+
+  /// Mark this render object's layout information as dirty
+  ///
+  /// Rather than eagerly updating layout information in response to writes into
+  /// this render object, we instead mark the layout information as dirty, which
+  /// schedules a visual update. As part of the visual update, the rendering
+  /// pipeline will update this render object's layout information.
+  ///
+  /// This mechanism batches the layout work so that multiple sequential writes
+  /// are coalesced, removing redundant computation.
```

Comment:
Also document how this impacts needsLayout.
Also document how this decides to go mark a parent as needsLayout.